app/forms/vpc-router-route-create.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useNavigate } from 'react-router-dom'
+
+import { useApiMutation, useApiQueryClient, type RouterRouteCreate } from '@oxide/api'
+
+import { DescriptionField } from '~/components/form/fields/DescriptionField'
+import { ListboxField } from '~/components/form/fields/ListboxField'
+import { NameField } from '~/components/form/fields/NameField'
+import { TextField } from '~/components/form/fields/TextField'
+import { SideModalForm } from '~/components/form/SideModalForm'
+import { fields, targetValueDescription } from '~/forms/vpc-router-route/shared'
+import { useForm, useVpcRouterSelector } from '~/hooks'
+import { addToast } from '~/stores/toast'
+import { pb } from '~/util/path-builder'
+
+const defaultValues: RouterRouteCreate = {
+  name: '',
+  description: '',
+  destination: { type: 'ip', value: '' },
+  target: { type: 'ip', value: '' },
+}
+
+export function CreateRouterRouteSideModalForm() {
+  const queryClient = useApiQueryClient()
+  const routerSelector = useVpcRouterSelector()
+  const navigate = useNavigate()
+
+  const onDismiss = () => {
+    navigate(pb.vpcRouter(routerSelector))
+  }
+
+  const createRouterRoute = useApiMutation('vpcRouterRouteCreate', {
+    onSuccess() {
+      queryClient.invalidateQueries('vpcRouterRouteList')
+      addToast({ content: 'Your route has been created' })
+      onDismiss()
+    },
+  })
+
+  const form = useForm({ defaultValues })
+  const targetType = form.watch('target.type')
+
+  const onChangeTargetType = (value: string | null | undefined) => {
```

Comment:
The nice thing about inlining callbacks like this is you don't have to annotate the argument type

Reply:
Great point!